docs/book_examples/src/widget_md.rs

```diff
@@ -0,0 +1,73 @@
+use druid::Data;
+
+// ANCHOR: padded_button
+use druid::widget::{Button, Padding, Widget};
+
+fn padded_button<T: Data>() -> impl Widget<T> {
+    let button = Button::new("Humour me", Button::noop);
+    Padding::new((4.0, 8.0), button)
+}
+// ANCHOR_END: padded_button
+
+// ANCHOR: align_center
+use druid::widget::Align;
+
+fn align_center<T: Data>(widget: impl Widget<T> + 'static) -> impl Widget<T> {
+    Align::centered(widget)
+}
+// ANCHOR_END: align_center
+
+// ANCHOR: stepper_builder
+use druid::widget::Stepper;
+
+fn steppers() {
+    // a Stepper with default paramaters
+    let stepper1 = Stepper::new();
+
+    // A Stepper that operates over a custom range
+    let stepper2 = Stepper::new().with_range(10.0, 50.0);
+
+    // A Stepper with a custom range *and* a custom step size, that
+    // wraps around past its min and max values:
+    let stepper3 = Stepper::new()
+        .with_range(10.0, 50.0)
+        .with_step(2.5)
+        .with_wraparound(true);
+}
+// ANCHOR_END: stepper_builder
+
+#[rustfmt::skip]
+mod padded_stepper_raw {
+// ANCHOR: padded_stepper_raw
+use druid::widget::{Align, Padding, Stepper};
+
+fn padded_stepper() {
```

Comment:
The true function signature is the real gotcha (in my experience) of widget composition, so I think it would be nice to have it even in simple samples like this.

Reply:
To be clear, that part that wasn't obvious to me is:
```rust
fn align_center<T: Data>
```
The idea that my function has this type... I actually don't know the word for it. It's sort of this invisible extra parameter being passed to the function.

Reply:
I agree, I think the issue here is that we're wrapping a widget, which means we need to know what T is; if we don't it becomes a parameter of the function. I think the best thing to do here is to just not have these return anything.